include/swift/AST/DiagnosticsSIL.def

```diff
@@ -753,50 +753,32 @@ ERROR(sil_moveonlychecker_owned_value_consumed_and_used_at_same_time, none,
 ERROR(sil_moveonlychecker_value_used_after_consume, none,
       "'%0' used after consume", (StringRef))
 ERROR(sil_moveonlychecker_guaranteed_value_consumed, none,
-      "'%0' has guaranteed ownership but was consumed", (StringRef))
+      "'%0' is borrowed and cannot be consumed", (StringRef))
+
+// FIXME: this diagnostic shouldn't ever be emitted now. rdar://109742587 (closures may still try to consume captures, e.g., borrowed parameters)
 ERROR(sil_moveonlychecker_guaranteed_value_captured_by_closure, none,
-      "'%0' has guaranteed ownership but was consumed due to being captured by a closure", (StringRef))
-ERROR(sil_moveonlychecker_let_value_consumed_in_closure, none,
-      "'%0' consumed in closure. This is illegal since if the closure is invoked more than once the binding will be uninitialized on later invocations", (StringRef))
+      "'%0' is borrowed and cannot be consumed by closure capture", (StringRef))
+
+ERROR(sil_moveonlychecker_capture_consumed_in_closure, none,
+      "captured '%0' cannot be consumed within a closure", (StringRef))
 ERROR(sil_moveonlychecker_inout_not_reinitialized_before_end_of_function, none,
-      "'%0' consumed but not reinitialized before end of function", (StringRef))
-ERROR(sil_moveonlychecker_inout_not_reinitialized_before_end_of_closure, none,
-      "'%0' consumed in closure but not reinitialized before end of closure", (StringRef))
+      "missing reinitialization of '%0' after consume", (StringRef))
```

Comment:
By removing the information explaining to the user that the value needs to be reinitialized before the end of the function, the error is no longer actionable for the user unless they understand the properties of inout. Can you add that information back in?